### ISSUE TITLE
Usage of ChipTAN?

### ISSUE THREAD

**Issue by flyinggreenfrog** [grindsa/dkb-robo#56]
I just saw your project and wanted to try it out. Installing it via pip3, and then using the CLI with ChipTAN (as I don't use any form of the DKB Apps). From the documentation it's not clear to me how the `dkb` CLI works with ChipTAN. It should, if I understand your documentation correctly: `-t, --use-tan dbk-robo will ask for a TAN (generated by either ChipTan or TAN2go) during login`.

How is the QR Code of the ChipTAN process shown (opened in a separate browser window?) or how does dkb_robo work with ChipTAN.

If I try to use without `--use-tan` option, then it uses `https://banking.dkb.de:443`, but no mfa_devices are found?

If I use with `--use-tan` option, it uses `https://www.ib.dkb.de:443` (which DKB will shutdown in September?), then I got some error `confirmForm not found`.

I try to use as follows: `dkb --debug -u XXX -p XXX --use-tan accounts`.

**Comment by grindsa** [comment]
I did not test ChipTan as it was not supported in the new UI. Seems its supported now. So let me have a look and try to find my old reader.... For the moment should only work in scenarios where you have to insert a TAN ("chipTAN manuell").

**Comment by grindsa** [comment]
Hi,

I added support for [chipTAN manual](https://www.dkb.de/fragen-antworten/was-ist-das-chiptan-verfahren) to the [devel branch](https://www.dkb.de/fragen-antworten/was-ist-das-chiptan-verfahren) . Functionality is similar to the legacy implementation, meaning you need to add the TAN manually.

```bash
dkb -u <dkb_user> -p <dkb_pasword> -t accounts

Using to chip_tan to login
Du hast das Verfahren chipTAN manuell ausgewählt.

1. Stecke die Karte (･･････ xxx) in den TAN-Generator und drücke die Taste TAN.

2. Gib den Startcode 12345678 ein und drücke die Taste OK.

3. Bitte bestätige den Auftrag mit der TAN deines TAN-Generators.

TAN: 651423
...
```
Feel free to test. I will look into chipTAN QR next...

**Comment by grindsa** [comment]
Hi again,

I did a proof-of-concept to implement chipTAN-qr support.  I am able to get the QR code however I did not find a good way to display the QR code on console level. Decoding by [pyzbar](https://github.com/NaturalHistoryMuseum/pyzbar/blob/master/pyzbar/pyzbar.py) or [cv2](https://github.com/opencv/opencv-python) and recoding by using [qrcode](https://github.com/lincolnloop/python-qrcode) does not work as the qrcode from DKB contains binary data so decoding fails. 

Thus, I decided to use python-pillow do display the QR code in a separate window opened by the default image viewer.

Here is an example from my system.  Would this be an acceptable way forward?

![image](https://github.com/user-attachments/assets/f0801b2a-bd93-4b48-9d86-b3c0280fee8b)


Feel free to try it yourself. The changes got pushed into a [separate branch](https://github.com/grindsa/dkb-robo/tree/chip_tan). You need to download and install it and install pillow via pip.

Feedback is always welcome…

G.

**Comment by flyinggreenfrog** [comment]
Thank you so much for your efforts so far, it's really appreciated! Unfortunately it may take some time on my side to test it, just wanted to let you know and thank you for your work. Apologies already in delay of testing on my side that I foresee because of different reasons on my side.

**Comment by grindsa** [comment]
no problem at all. I released the fixes as part of [dkb-robo v0.27](https://github.com/grindsa/dkb-robo/releases/tag/0.27). So please use this release for testing. I am closing the issue. If you face problems during testing pls reopen.
